src/mame/layout/mpu4.lay

```diff
@@ -30,6 +30,264 @@ license:CC0
 		</led16segsc>
 	</element>
 
+	
+	<element name="text0"><text string="0"><color red="0.03" green="0.03" blue="0.03" /></text></element>
+	<element name="text1"><text string="1"><color red="0.03" green="0.03" blue="0.03" /></text></element>
+	<element name="text2"><text string="2"><color red="0.03" green="0.03" blue="0.03" /></text></element>
+	<element name="text3"><text string="3"><color red="0.03" green="0.03" blue="0.03" /></text></element>
```

Comment:
MAME does allow generating these elements in a loop.  Fore example see here: https://github.com/mamedev/mame/blob/mame0244/src/mame/layout/intlc44.lay#L45

Reply:
I didn't know that, although in terms of practical application I wonder what is actually best.

If you're trying to figure out a layout, it helps to be able to change the text on the debug display, which obviously you can't do once you put it in a loop.

Reply:
No, the format’s effectively declarative, you can’t change the element text after it’s created.  If you look at the some of the more fleshed-out fruit machine layouts, they have a separate debug display with numbered lights with labels generated using loops in addition to the live display.

For example if you look at m5arab.lay:
* The loops that generate the labels for the numbered debug lamps and buttons are at lines 2322 and 2347, respectively.
* The “debug” view starts at line 3099.
* The actual lamps are not added to the view using loops because different colour elements are used for lamps that are unused by the game (`debug_lamp_unreferenced`), used for regular lamps (`debug_lamp_standard`), used for illuminated buttons (`debug_lamp_button`) or used for multi-segment display segments (`debug_lamp_segment`).
* The labels are overlaid on the lamps using loops at line 3873.
* The buttons in the debug view start after this.  Similarly, they aren’t in a loop because they use different colours depending on how the game uses (or doesn’t use) them.  In this particular layout, they don’t have their port tags set yet.
* The button number are overlaid with a loop at line 4078.

The debug views are pretty similar across the layouts for games on the same platform – the main thing that varies is the colour of the lamps/buttons as that reflects how the game uses them.